src/server/sched_func.c

```diff
@@ -331,13 +381,28 @@ int
 poke_scheduler(attribute *pattr, void *pobj, int actmode)
 {
 	if (pobj == &server || pobj == dflt_scheduler) {
+		if (pobj == &server) {
+			/* set this attribute on main scheduler */
+			if (dflt_scheduler) {
+				dflt_scheduler->sch_attr[SCHED_ATR_scheduling].at_val.at_long = pattr->at_val.at_long;
+				dflt_scheduler->sch_attr[SCHED_ATR_scheduling].at_flags |=
+						ATR_VFLAG_SET | ATR_VFLAG_MODIFY | ATR_VFLAG_MODCACHE;
+				(void)sched_save_db(dflt_scheduler, SVR_SAVE_FULL);
+			}
+		} else {
+			server.sv_attr[SRV_ATR_scheduling].at_val.at_long = pattr->at_val.at_long;
+			server.sv_attr[SRV_ATR_scheduling].at_flags |= ATR_VFLAG_MODCACHE;
+			svr_save_db(&server, SVR_SAVE_QUICK);
+		}
 		if (actmode == ATR_ACTION_ALTER) {
 			if (pattr->at_val.at_long)
-				set_scheduler_flag(SCH_SCHEDULE_CMD);
+				set_scheduler_flag(SCH_SCHEDULE_CMD, dflt_scheduler);
 		}
-		/*TODO also set this attribute on main scheduler */
 	} else {
-		/*TODO need add the code for other scheduler apart from main scheduler. */
+		if (actmode == ATR_ACTION_ALTER) {
+			if (pattr->at_val.at_long)
```

Comment:
Why do we check for a value? Does this mean that one can not set the scheduling attribute on a multisched to 0?

Reply:
We only need to set the flag SCH_SCHEDULE_CMD if scheduling is set to true. To answer your question yes one can set scheduling of multisched to false if they want to turn off scheduling it works fine.